samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -188,6 +193,7 @@ public void onNewJobModelConfirmed(String version) {
   private void generateNewJobModel() {
     // get the current list of processors
     List<String> currentProcessors = zkUtils.getSortedActiveProcessors();
+    List<String> currentProcessorsPids = zkUtils.getSortedActiveProcessorsPIDs();
```

Comment:
yeah, same as getSortedActiveProcessors() today. 
I am not sure it can be handled here. It requires handling at 'debounce' scheduler level.